etl/steps/data/garden/ember/2022-12-13/yearly_electricity.py

```diff
@@ -0,0 +1,387 @@
+
+    Returns
+    -------
+    table : catalog.Table
+        Table of processed data for the given category.
+
+    """
+    # Prepare wide table of emissions data.
+    table = prepare_wide_table(df=df, category="Power sector emissions")
+
+    # Add carbon intensity.
+    # In principle this only needs to be done for region aggregates, but we do it for all countries and check that
+    # the results are consistent with the original data.
+    # Prepare wide table also for electricity generation (required to calculate carbon intensity).
+    electricity = prepare_wide_table(df=df, category="Electricity generation")[
+        ["country", "year", "Total Generation (TWh)"]
+    ]
+    # Add total electricity generation to emissions table.
+    table = pd.merge(table, electricity, on=["country", "year"], how="left")
+    # Rename the original carbon intensity column as a temporary column called "check".
+    intensity_col = "CO2 intensity (gCO2/kWh)"
+    table = table.rename(columns={intensity_col: "check"})
+    # Calculate carbon intensity for all countries and regions.
+    table[intensity_col] = (
+        pd.DataFrame(table)["Total emissions (mtCO2)"] * MT_TO_G / (table["Total Generation (TWh)"] * TWH_TO_KWH)
+    )
+
+    # Check that the new carbon intensities agree (within 1 % of mean average percentage error) with the original
+    # ones (where carbon intensity was given, namely for countries, not aggregate regions).
+    mape = 100 * abs(table.dropna(subset="check")[intensity_col] - table["check"].dropna()) / table["check"].dropna()
+    assert mape.max() < 1, "Calculated carbon intensities differ from original ones by more than 1 percent."
+
+    # Remove temporary column.
+    table = table.drop(columns="check")
+
+    return table
+
+
+def run(dest_dir: str) -> None:
+    log.info(f"{DATASET_SHORT_NAME}.start")
+
+    #
+    # Load data.
+    #
+    # Read dataset from meadow.
+    ds_meadow = catalog.Dataset(MEADOW_DATASET_PATH)
+    # Get table from dataset.
+    tb_meadow = ds_meadow[DATASET_SHORT_NAME]
+    # Make a dataframe out of the data in the table, with the required columns.
+    df = load_global_electricity_review(tb_meadow)
+
+    #
+    # Process data.
+    #
+    # Harmonize country names.
+    log.info(f"{DATASET_SHORT_NAME}.harmonize_countries")
+    df = geo.harmonize_countries(df=df, countries_file=str(COUNTRY_MAPPING_PATH))
+
+    # Split data into different tables, one per category, and process each one individually.
+    log.info(f"{DATASET_SHORT_NAME}.prepare_wide_tables")
+    tables = {
+        "Capacity": prepare_wide_table(df=df, category="Capacity"),
+        "Electricity demand": process_electricity_demand(df=df),
+        "Electricity generation": process_electricity_generation(df=df),
+        "Electricity imports": prepare_wide_table(df=df, category="Electricity imports"),
+        "Power sector emissions": process_power_sector_emissions(df=df),
```

Comment:
I would probably standardise the names of the functions, to make it obvious that _all_ of them output a table.

E.g. something like 

- `process_electricity_demand` → `make_table_electricity_demand`
- `process_electricity_generation` → `make_table_electricity_generation`
- `process_power_sector_emissions` → `make_table_power_sector_emissions`. 

And for consistency, `prepare_wide_table` → `make_wide_table`

Reply:
Done.